docs/tutorial/en/source/tutorial/rag.py

```diff
@@ -0,0 +1,294 @@
+            "config_name": "qwen_emb_config",
+            "model_name": "text-embedding-v2",
+            "api_key": os.getenv("DASHSCOPE_API_KEY"),
+        },
+    ],
+)
+
+local_knowledge = LlamaIndexKnowledge.build_knowledge_instance(
+    knowledge_id="agentscope_qa",
+    data_dirs_and_types={"./": [".md"]},
+    emb_model_name="qwen_emb_config",
+)
+
+
+nodes = local_knowledge.retrieve(
+    "what is agentscope?",
+    similarity_top_k=1,
+)
+
+print()
+print("The retrieved content:")
+nodes[0].node.get_content()
+
+# %%
+# If one wants to have more control on how the data are preprocessing,
+# a knowledge configuration can be passed to the function.
+# Especially, `SimpleDirectoryReader` is the class in LlamaIndex library, and `init_args` is the initialization parameters of `SimpleDirectoryReader`.
+# As for the data preprocessing, developers can choose different LlamaIndex `transformation operations <https://docs.llamaindex.ai/en/stable/module_guides/loading/ingestion_pipeline/transformations/>`_ to preprocess the data.
+
+flex_knowledge_config = {
+    "knowledge_id": "agentscope_qa_flex",
+    "knowlege_type": "llamaindex_knowledge",
+    "emb_model_config_name": "qwen_emb_config",
+    "chunk_size": 1024,
+    "chunk_overlap": 40,
+    "data_processing": [
+        {
+            "load_data": {
+                "loader": {
+                    "create_object": True,
+                    "module": "llama_index.core",
+                    "class": "SimpleDirectoryReader",
+                    "init_args": {
+                        "input_dir": "./",
+                        "required_exts": [
+                            ".md",
+                        ],
+                    },
+                },
+            },
+            "store_and_index": {
+                "transformations": [
+                    {
+                        "create_object": True,
+                        "module": "llama_index.core.node_parser",
+                        "class": "SentenceSplitter",
+                        "init_args": {
+                            "chunk_size": 1024,
+                        },
+                    },
+                ],
+            },
+        },
+    ],
+}
+
+local_knowledge_flex = LlamaIndexKnowledge.build_knowledge_instance(
+    knowledge_id="agentscope_qa_flex",
+    knowledge_config=flex_knowledge_config,
+)
+
+
+nodes = local_knowledge.retrieve(
+    "what is agentscope?",
+    similarity_top_k=1,
+)
+
+print()
+print("The retrieved content:")
+nodes[0].node.get_content()
+
+# %%
+# Create a `BingKnowledge` instance
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# If one want to have online search engine for RAG application, `BingKnowledge` can be an option.
+# `BingKnowledge` requires a Bing search API key .
+
+from agentscope.rag.search_knowledge import BingKnowledge
+
+bing_key = os.getenv("BING_SEARCH_KEY")
+online_knowledge = BingKnowledge.build_knowledge_instance(
```

Comment:
What's the difference between the service bing search and bing search knowledge?